src/pages/AddItemPage.tsx

```diff
@@ -0,0 +1,22 @@
+import { useNavigate } from "react-router-dom";
+import ItemForm from "../components/ItemForm";
+import useApi, { Item } from "../hooks/useApi";
+import { endpoint } from "../assets/constants/constants";
+
+const AddItemPage = () => {
+  const { addItem } = useApi(endpoint);
+  const navigate = useNavigate();
+
+  const handleSubmit = async (data: Item) => {
+    try {
+      addItem(data);
+      navigate("/");
+    } catch (error) {
+      console.error("Error occured when trying to add element:", error);
```

Comment:
Same goes for the error handling. The mutation accepts an `onError` property, so we can move it there. But this is just a generic message, so no need to pass it to the hook, can just move it directly. Also, please have it translated and use an alert instead of console.log, so the end user has some feedback when something goes wrong.